### PR TITLE
Client: upscale [mask or base image

### LINKED ISSUE #14796: Fix ^[mask texture modifier
### Minetest version

```true
5.8
```

### Irrlicht device

_No response_

### Operating system and version

all

### CPU model

_No response_

### GPU model

_No response_

### Active renderer

_No response_

### Summary

Mods assume that texture1.png^[mask:texture2.png scales texture2.png to the size of texture1.png before masking.
This can be seen in Mobs Monsters. An egg mask from the mod is overlayed onto a texture from MTG. A 128x128 texture pack is used.
mesemonster

While the scaling can be done by mods, it is quite cumbersome:
```
local function mask_escape(str)
	return str:gsub("%^%[", "\\%^\\%["):gsub(":", "\\:")
end
"texture1.png^[mask:"..mask_escape("((texture1.png^[opacity:0)^texture2.png)"),
```
I wish for this to be fixed in 5.9. Shouldn't be hard as most other texture modifiers already use scaling.

A resolution-independent alternative to combine and resize would also be appreciated.

Mods shouldn't care about texture size because they run on the server and don't know anything about it and shouldn't make any assumptions about it. That's what I've been told on the Minetest IRC when I naively asked about how to get a texture's resolution. So stick to your dogma.

### Steps to reproduce

`texture1.png^[mask:texture2.png`

### PR DESCRIPTION
**Pull request by SmallJoker**
This improves texture pack compatibility. Masks are expected to be of the same size as the base texture. This change upscales the smaller texture if needed. The behaviour is now the same as `a.png^b.png` and `a.png^[overlay:b.png` (to mention a few).

Fixes #14796

## To do

This PR is Ready for Review.

## How to test

1. devtest node `testnodes:modifier_mask`
2. Place the node in master and with this PR

Expected outcome:

reference

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
Makes sense and shouldn't cause problems anywhere.

**Comment by grorp** [comment]
This needs documentation in lua_api.md

File: doc/lua_api.md [hunk 1]
```diff
@@ -490,6 +490,11 @@ to let the client generate textures on-the-fly.
 The modifiers are applied directly in sRGB colorspace,
 i.e. without gamma-correction.
 
+### Notes
+
+ * `TEXMOD_UPSCALE`: The texture with the lower resolution will be automatically
+   upscaled to the higher resolution texture.
```
Comment (grorp) [inline doc/lua_api.md:496]: In general I'd say that documentation is not code and making it human-readable is more important than keeping it "DRY", but this style of documentation seems fine here.

What about `[png`? It uses `blitBaseImage` which calls `upscaleImagesToMatchLargest`.

Comment (SmallJoker) [reply doc/lua_api.md:496]: Good catch. Documented that one too. I agree that the documentation should not be cryptic, but if I'd inline this part, the documentation would become even harder to maintain.

**Review by grorp — APPROVED** [review]

**Pull request merged**